src/elliptic/curves/mod.rs

```diff
@@ -1,6 +1,111 @@
+//! # General purpose elliptic curve cryptography
+//!
+//! Here we define generic elliptic curve cryptography and provide implementation of several curves.
+//!
+//! ## Usage
+//!
+//! Elliptic curve cryptography involves points and scalars. We provide respective structures
+//! [Point\<E\>](Point), [Scalar\<E\>](Scalar), where generic `E` stands for choice of elliptic
+//! curve, e.g. [Secp256k1] (`Point<Secp256k1>`, `Scalar<Secp256k1>`).
+//!
+//! Various methods and traits are defined for points and scalars which basically empowers you to do
+//! anything you can do in elliptic curve cryptography.
+//!
+//! ## Examples
+//!
+//! ### Public point/private scalar generation
+//! ```rust
+//! use curv::elliptic::curves::{Point, Scalar, Secp256k1};
+//!
+//! // Samples a random nonzero scalar (mod group order)
+//! let secret = Scalar::<Secp256k1>::random();
+//! // Multiplies generator at secret, retrieving a public point
+//! let public = Point::generator() * secret;    
+//! ```
+//!
+//! ### Diffie-Hellman
+//!
+//! Function below is a final step of the Diffie-Hellman key exchange protocol when both parties
+//! have exchanged their ephemeral public keys. Giving it here just for an example, the `curv` library
+//! is equipped with an implementation of this protocol (see [dh_key_exchange], or its more involved
+//! version: [dh_key_exchange_variant_with_pok_comm]).
+//!
+//! [dh_key_exchange]: crate::cryptographic_primitives::twoparty::dh_key_exchange
+//! [dh_key_exchange_variant_with_pok_comm]: crate::cryptographic_primitives::twoparty::dh_key_exchange_variant_with_pok_comm
+//!
+//! ```rust
+//! use curv::elliptic::curves::{Point, Scalar, Secp256k1};
+//!
+//! fn diffie_hellman(
+//!     my_secret: &Scalar<Secp256k1>,
+//!     counterparty_point: &Point<Secp256k1>
+//! ) -> Point<Secp256k1> {
+//!     my_secret * counterparty_point
+//! }
+//! ```
+//!
+//! You may have noticed that this function lacks of subgroup check (whether counterparty
```

Comment:
secp256k1 is a prime order group so in this specific case this check will be unnecessary

Reply:
Good point! At point of writing docs, there were no others curves implemented. I'll change the example to use ed25519.

Reply:
Fixed